src/computeSourceMap.ts

```diff
@@ -33,3 +51,60 @@ export default function computeSourceMap(
     names: [],
   };
 }
+
+function computeSourceColumns(code: string, tokens: Array<Token>): Array<number> {
+  const sourceColumns: Array<number> = new Array(tokens.length);
+  let j = 0;
+  let currentMapping = tokens[j].start;
+  let lineStart = 0;
+  for (let i = 0; i < code.length; i++) {
+    if (i === currentMapping) {
+      sourceColumns[j] = currentMapping - lineStart;
+      currentMapping = tokens[++j].start;
+    }
+    if (code.charCodeAt(i) === charCodes.lineFeed) {
+      lineStart = i + 1;
+    }
+  }
+  return sourceColumns;
+}
+
+function computeDetailedSourceMap(
+  code: string,
+  filePath: string,
+  {compiledFilename}: SourceMapOptions,
+  rawMappings: Array<number | undefined>,
+  sourceColumns: Array<number>,
+): RawSourceMap {
+  const map = new GenMapping({file: compiledFilename});
+  let j = 0;
+  let currentMapping = rawMappings[j];
+  for (; currentMapping === undefined; currentMapping = rawMappings[++j]);
```

Comment:
Could you add a bounds check on `j` in the condition here? As I was trying out different implementation options I ran into an infinite loop here if none of the tokens had mappings.